src/py/mat3ra/made/tools/build/slab/builders.py

```diff
@@ -29,79 +26,75 @@ class PymatgenSlabGeneratorParameters(PymatgenSlabGeneratorParametersSchema):
 
-        return materials_with_vacuum
+    def get_material(self, configuration: _ConfigurationType) -> Material:  # type: ignore
+        self._configuration = configuration
+        return super().get_material(configuration)
 
     def get_terminations(self, configuration: _ConfigurationType) -> List[Termination]:
-        return [
-            self.__create_termination_from_slab_pymatgen(slab)
-            for slab in self._generate_or_get_from_cache(configuration)
-        ]
-
-    def __conditionally_convert_slab_to_orthogonal_pymatgen(self, slab: PymatgenSlab) -> PymatgenSlab:
-        return slab.get_orthogonal_c_slab() if self.__configuration.use_orthogonal_z else slab
+        """Get available terminations for the given configuration."""
+        atomic_layers: AtomicLayersUniqueRepeated = configuration.stack_components[0]
+        return atomic_layers.get_terminations()
+
+    def _generate(self, configuration: _ConfigurationType) -> List[Material]:  # type: ignore
+        atomic_layers: AtomicLayersUniqueRepeated = configuration.stack_components[0]
+        vacuum: VacuumConfiguration = configuration.stack_components[1]
+        params = self.build_parameters or SlabBuilderParameters()
+
+        if params.use_orthogonal_c:
+            pymatgen_slabs = atomic_layers._generate_pymatgen_slabs(
+                min_vacuum_size=params.min_vacuum_size,
+                in_unit_planes=True,
+                make_primitive=params.make_primitive,
+                symmetrize=params.symmetrize,
+            )
+            orthogonal_slabs = [slab.get_orthogonal_c_slab() for slab in pymatgen_slabs]
+            slab_materials = [Material.create(from_pymatgen(slab)) for slab in orthogonal_slabs]
+        else:
+            slab_materials = atomic_layers.get_slabs(
+                min_vacuum_size=params.min_vacuum_size,
+                in_unit_planes=True,
+                make_primitive=params.make_primitive,
+                symmetrize=params.symmetrize,
+            )
+
+        stacked_materials = []
+        for slab_material in slab_materials:
+            stacked = stack_two_components(slab_material, vacuum, direction=configuration.direction)
+            stacked_materials.append(stacked)
+
+        return stacked_materials
+
+    def _post_process(self, items: List[Material], post_process_parameters=None) -> List[Material]:
+        materials = items
+        materials = [create_supercell(material, self._configuration.xy_supercell_matrix) for material in materials]
+
+        for material in materials:
+            if "build" not in material.metadata:
+                material.metadata["build"] = {}
+            atomic_layers = self._configuration.stack_components[0]
+            material.metadata["build"]["termination"] = str(atomic_layers.termination_top)
+            material.metadata["build"]["configuration"] = self._configuration.to_dict()
+            material.metadata["build"]["build_parameters"] = (
```

Comment:
We can have `to_metadata` in Configuration to return this and just use material.metadata = `self._configuration.to_metadata()`